Backend/app/spotify_electron/health/health_controller.py

```diff
@@ -3,19 +3,119 @@
 """
 
 from fastapi import APIRouter
-from fastapi.responses import Response
-from starlette.status import HTTP_200_OK
+from fastapi.responses import JSONResponse
+from starlette.status import HTTP_200_OK, HTTP_503_SERVICE_UNAVAILABLE
+
+from app.auth.auth_schema import AuthConfig
+from app.database.database_schema import DatabasePingFailedException
+from app.database.DatabaseConnectionManager import DatabaseConnectionManager
+from app.logging.logging_constants import LOGGING_HEALTH
+from app.logging.logging_schema import SpotifyElectronLogger
+from app.spotify_electron.health.health_schema import HealthCheckResponse
+from app.spotify_electron.song.providers.song_service_provider import SongServiceProvider
 
 router = APIRouter(prefix="/health", tags=["health"])
+logger = SpotifyElectronLogger(LOGGING_HEALTH).getLogger()
+
+
+def check_database_connection() -> dict[str, str]:
```

Comment:
This can return None if the `if` statement is not met. We should change ping database to throw an exception if ping couldn't be done instead of returning a bool. I still feel like this func belongs more to a service layer.